### ISSUE TITLE
Error in email editor when a user does not have 'viewother' permissions

### ISSUE THREAD

**Issue by pjrobertson** [mautic/mautic#13810]
### Mautic Version

5.0.x series

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.0.30

### What happened?

1. Create a role called 'own documents' that only gives permissions for the user to 'view/edit/delete/create own' (no 'view other' permissions)
2. Create a user with the 'own documents' role
3. Login as this user, and attempt to create and edit an email. The email field is blank, even if you enter text:

https://github.com/mautic/mautic/assets/150431/e3d84fa0-129a-41a1-942b-b71f642f3ab7

Upon investigation, this is what is in my log files:

```
[2024-06-04T09:55:35.766922+00:00] mautic.CRITICAL: Uncaught PHP Exception Error: "Call to a member function with() on null" at /path/public/app/bundles/CoreBundle/Helper/BuilderTokenHelper.php line 99 {"exception":"[object] (Error(code: 0): Call to a member function with() on null at /path/public/app/bundles/CoreBundle/Helper/BuilderTokenHelper.php:99)"} {"hostname":"sjcx.smartairfilters.com","pid":3342775}
```

Changing the code to check for `null` first fixes the bug.

### How can we reproduce this issue?

Step 1: 

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by RCheesley** [comment]
Thanks for reporting the bug @pjrobertson and double thanks for the PR that fixes it! Hopefully we can get that tested and merged into the next release! 🚀

**Comment by chergett** [comment]
This is still relevant, and the bug still exists in version 6.0.0. I verified the fix above is still good in version 6.0.0.
